src/apolo_app_types/protocols/dify.py

```diff
@@ -42,3 +53,178 @@ class DifyOutputs(AppOutputsDeployer):
+        ...,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Master Preset",
+            description="Configure Master Preset.",
+        ).as_json_schema_extra(),
+    )
+
+
+class DifyAppInputs(AppInputs):
+    ingress_http: IngressHttp | None = Field(
+        default_factory=lambda: IngressHttp(),
+        json_schema_extra=SchemaExtraMetadata(
+            title="HTTP Ingress",
+            description="Define HTTP ingress configuration"
+            " for exposing services over the web.",
+        ).as_json_schema_extra(),
+    )
+    api: DifyAppApi
+    worker: DifyAppWorker
+    proxy: DifyAppProxy
+    web: DifyAppWeb
+    redis: DifyAppRedis
+    bucket: Bucket = Field(
+        ...,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Bucket",
+            description="Configure Dify Blob Storage (Bucket).",
+        ).as_json_schema_extra(),
+    )
+    external_postgres: CrunchyPostgresUserCredentials
+    external_pgvector: CrunchyPostgresUserCredentials
+
+
+class DifySpecificOutputs(BaseModel):
+    model_config = ConfigDict(
+        protected_namespaces=(),
+        json_schema_extra=SchemaExtraMetadata(
+            title="Dify Specific Outputs",
+            description="Configure Dify Specific Outputs.",
+            meta_type=SchemaMetaType.INTEGRATION,
+        ).as_json_schema_extra(),
+    )
+    init_password: str = Field(
+        ...,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Init Password",
+            description="The initial password for the Dify application.",
+        ).as_json_schema_extra(),
+    )
+
+
+class DifyAppOutputs(AppOutputs):
+    internal_web_app_url: RestAPI | None = Field(
+        None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Internal Web App URL",
+            description="The URL of the internal web app.",
+        ).as_json_schema_extra(),
+    )
+    external_web_app_url: RestAPI | None = Field(
+        default=None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="External Web App URL",
+            description="The URL of the external web app.",
+        ).as_json_schema_extra(),
+    )
+    internal_api_url: RestAPI | None = Field(
+        default=None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Internal API URL",
+            description="The URL of the internal API.",
+        ).as_json_schema_extra(),
+    )
+    external_api_url: RestAPI | None = Field(
+        default=None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="External API URL",
+            description="The URL of the external API.",
+        ).as_json_schema_extra(),
+    )
+    dify_specific: DifySpecificOutputs | None = Field(
+        default=None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Dify Specific Outputs",
+            description="The URL of the external API.",
+        ).as_json_schema_extra(),
+    )
```

Comment:
since this is required

```suggestion
    dify_specific: DifySpecificOutputs
```